From sshtylyov@ru.mvista.com  Tue May  5 09:23:41 2009
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Date: Tue, 5 May 2009 15:34:34 +0400
Subject: cs5536: define dma_sff_read_status() method
To: stable@kernel.org
Message-ID: <200905051534.34672.sshtylyov@ru.mvista.com>

From: Sergei Shtylyov <sshtylyov@ru.mvista.com>

commit 15da90b516e9da92cc1d90001e640fd6707d0e27 upstream.

The driver somehow got merged with the initializer for the dma_sff_read_status()
method missing which caused kernel panic on bootup.

This should fix the kernel.org bug #13026...

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Reported-by: Arnd Hannemann <hannemann@nets.rwth-aachen.de>
Signed-off-by: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/ide/cs5536.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/ide/cs5536.c
+++ b/drivers/ide/cs5536.c
@@ -237,6 +237,7 @@ static const struct ide_dma_ops cs5536_d
 	.dma_test_irq		= ide_dma_test_irq,
 	.dma_lost_irq		= ide_dma_lost_irq,
 	.dma_timeout		= ide_dma_timeout,
+	.dma_sff_read_status	= ide_dma_sff_read_status,
 };
 
 static const struct ide_port_info cs5536_info = {
